tests/test_providers.py

```diff
@@ -263,3 +275,464 @@ def parse_(self, response: DummyResponse, page: AnnotatedProductPage):  # type:
+    class SomePage(BasePage):
+        response: AnyResponse
+        browser_response: BrowserResponse
+        html: BrowserHtml
+
+    class ZyteAPISpider(Spider):
+        def start_requests(self):
+            yield Request(self.url, callback=self.parse_)
+
+        def parse_(self, response: DummyResponse, page: SomePage):
+            yield {"page": page}
+
+    settings = provider_settings(mockserver)
+    item, url, crawler = await crawl_single_item(ZyteAPISpider, HtmlResource, settings)
+    params = crawler.engine.downloader.handlers._handlers["http"].params
+
+    assert len(params) == 1
+    assert params[0].keys() == {"url", "browserHtml"}
+
+    assert type(item["page"].response) == AnyResponse
+    assert type(item["page"].response.response) == BrowserResponse
+    assert type(item["page"].browser_response) == BrowserResponse
+    assert type(item["page"].html) == BrowserHtml
+
+
+@ensureDeferred
+async def test_provider_any_response_http_response(mockserver):
+    @attrs.define
+    class SomePage(BasePage):
+        response: AnyResponse
+        http_response: HttpResponse
+
+    class ZyteAPISpider(Spider):
+        def start_requests(self):
+            yield Request(self.url, callback=self.parse_)
+
+        def parse_(self, response: DummyResponse, page: SomePage):
+            yield {"page": page}
+
+    settings = provider_settings(mockserver)
+    item, url, crawler = await crawl_single_item(ZyteAPISpider, HtmlResource, settings)
+    params = crawler.engine.downloader.handlers._handlers["http"].params
+
+    assert len(params) == 1
+    assert params[0].keys() == {
+        "url",
+        "httpResponseBody",
+        "customHttpRequestHeaders",
+        "httpResponseHeaders",
+    }
+
+    assert type(item["page"].response) == AnyResponse
+    assert type(item["page"].response.response) == HttpResponse
+    assert type(item["page"].http_response) == HttpResponse
+
+
+@ensureDeferred
+async def test_provider_any_response_browser_http_response(mockserver):
+    @attrs.define
+    class SomePage(BasePage):
+        response: AnyResponse
+        browser_response: BrowserResponse
+        http_response: HttpResponse
+
+    class ZyteAPISpider(Spider):
+        def start_requests(self):
+            yield Request(self.url, callback=self.parse_)
+
+        def parse_(self, response: DummyResponse, page: SomePage):
+            yield {"page": page}
+
+    settings = provider_settings(mockserver)
+    item, url, crawler = await crawl_single_item(ZyteAPISpider, HtmlResource, settings)
+    params = crawler.engine.downloader.handlers._handlers["http"].params
+
+    assert len(params) == 2
+    assert params[0].keys() == {
+        "url",
+        "httpResponseBody",
+        "customHttpRequestHeaders",
+        "httpResponseHeaders",
+    }
+    assert params[1].keys() == {"url", "browserHtml"}
+
+    assert type(item["page"].response) == AnyResponse
+    assert type(item["page"].response.response) == HttpResponse
```

Comment:
FIXME: Perhaps when `HttpResponse` and `BrowserResponse` are both available, `BrowserResponse` takes precedence.

Reply:
Done in https://github.com/scrapy-plugins/scrapy-zyte-api/pull/161/commits/85f355d3d91a6c4713746f12b9f8dbd96835405e